src/js/stores/document.js

```diff
@@ -169,7 +169,17 @@ define(function (require, exports, module) {
 
             this._openDocuments[nextDocument.id] = nextDocument;
 
-            if (!suppressChange) {
+            if (suppressChange) {
+                return;
+            }
+
+            // If some selected layer remains uninitialized, a new document will
+            // come along shortly. Wait until then to trigger the change event.
+            var initialized = nextDocument.layers.selected.every(function (layer) {
```

Comment:
I definitely understand/undertsood  *why* we wanted to limit the change events, I just wondered if there was a way to handle this in the document/layer action dances instead of inside the store.  But, I won't press too hard.  If there are issues, I'm sure they'll be immediate and obvious :eyes: